contrib/untar.c

```diff
@@ -133,18 +144,19 @@ verify_checksum(const char *p)
 static void
 untar(FILE *a, const char *path)
 {
-	char buff[512];
+#define BUF_SIZE 512
```

Comment:
Style: a #define should not be placed inside a function
If its only for this one function we cen declare a const variable

Reply:
But that will create a VLA.

Reply:
But that will create a VLA. Also it means I can't use the new-as-of-C89 feature of automatically concatenating literal strings together when they're next to each other. See A.2.6 of K&R (2nd edition):
> Adjacent string literals are concatenated into a single string.
[…]
> The specification that string literals need not be distinct, and the prohibition against modifying them, are new in the ANSI standard, as is the concatenation of adjacent string literals.

Reply:
Have to agree here - even with C99+ compilers - depending on how strict the settings are the compiler will create a VLA with both `const int buf_size = 512` and `static const int buf_size = 512`.

The `#define` seems odd, but having it scoped in the function is more portable and prevents collision.

Reply:
Not an ideal approach, but may be the best you can do here.  (This is C, after all. ;-)